packages/core/src/internal/channel.js

```diff
@@ -210,15 +210,38 @@ export function multicastChannel() {
   }
 }
 
-export function stdChannel() {
-  const chan = multicastChannel()
-  const { put } = chan
-  chan.put = input => {
-    if (input[SAGA_ACTION]) {
-      put(input)
-      return
-    }
+const scheduleSagaPut = put => input => {
+  if (input[SAGA_ACTION]) {
+    put(input)
+  } else {
     asap(() => put(input))
   }
+}
+
+const wrapSagaDispatch = dispatch => action => dispatch(Object.defineProperty(action, SAGA_ACTION, { value: true }))
+
+function liftable(chan) {
+  chan.lift = fn => {
+    if (process.env.NODE_ENV === 'development') {
+      check(fn, is.func(fn), 'channel.lift(fn): fn must be a function')
+    }
+    chan.put = fn(chan.put)
+    if (process.env.NODE_ENV === 'development') {
+      check(fn, is.func(chan.put), 'channel.lift(fn): fn must return a function')
+    }
+    return chan
+  }
+
+  chan._clone = () => {
+    const { put, take, close } = chan
+    return liftable({ [MULTICAST]: true, put, take, close })
```

Comment:
why using `clone` assume that we're using multicast channel?

Reply:
Only stdChannels have `lift / _clone / _connect` methods.  These methods are only used by stdChannel to monkey-patching its `put` and handle the scheduling of external IO.
Multicast channels or normal channels do not need to do such monkey-patching or scheduling, so they do not have these methods.

Reply:
I see. but the current implementation doesn't tell us the only stdChannel could be passed in.

Reply:
I agree this is not obvious that only stdChannel could be passed to `liftable`.  But `liftable` is internal (only `lift` method on stdChannel instance is public), so I think it's just fine.

Reply:
we have other `channels` defined in this file. probably we can just rename method to makeLiftableStdChannel